server/realTime/scripts/Network.js

```diff
@@ -1072,6 +1140,130 @@ Protocol.prototype.bindIO = function bindIO () {
 			});
 		});
 
+		socket.on("removeprotectedregion", function (regionId, callback) {
+			if (typeof callback !== 'function') return;
+
+			if(isNaN(regionId))
+			{
+				callback("Region id is undefined.")
+				return;
+			}
+
+			protocol.players.request('removeprotectedregion', {
+				uKey: socket.uKey,
+				room: socket.room,
+				regionId: regionId
+			}, function (err, data) {
+				callback(err, data);
+				protocol.updateProtectedRegions(socket.room);
+			});
+		});
+
+		socket.on("getmyprotectedregions", function (callback) {
+			if(!socket.userid)
```

Comment:
i'm not certain why i stringified it. I use the realtime's `protocol.getUsersProtectedRegions(socket.userid, socket.room)` which uses the local variable `this.protectedRegions[room]`. i figured that using the local variable that has the permissions already would be better than a database hit that would need to be joined between users, permissions, and regions.